_includes/comments-providers/discourse.html

```diff
@@ -1,7 +1,7 @@
 {% if site.comments.discourse.server %}
-{% capture canonical %}{% if site.permalink contains '.html' %}{{ page.url | absolute_url }}{% else %}{{ page.url | absolute_url | remove:'index.html' | strip_slash }}{% endif %}{% endcapture %}
+{% capture canonical %}{{ page.url | replace: 'index.html','' | prepend: site.baseurl | prepend: site.url }}{% endcapture %}
```

Comment:
Sure, but that's pretty much what it is now... except your version removes a few more checks.

I guess I'm waiting for someone else to verify this is actually broken and not outputting the expected URL for Discourse and not some sort of Jekyll/environment/configuration issue.